tests/acceptance/features/bootstrap/GraphContext.php

```diff
@@ -234,25 +234,49 @@ public function adminDeletesUserUsingTheGraphApi(string $user): void {
 	}
 
 	/**
-	 * @param string $user
+	 * remove user from group
+	 *
 	 * @param string $group
+	 * @param string $user
+	 * @param string|null $byUser
 	 *
-	 * @return void
-	 * @throws JsonException
+	 * @return ResponseInterface
 	 * @throws GuzzleException
 	 */
-	public function adminHasRemovedUserFromGroupUsingTheGraphApi(string $user, string $group): void {
-		$user = $this->featureContext->getActualUsername($user);
-		$userId = $this->featureContext->getAttributeOfCreatedUser($user, "id");
-		$groupId = $this->featureContext->getAttributeOfCreatedGroup($group, "id");
-		$response = GraphHelper::removeUserFromGroup(
+	public function removeUserFromGroup(string $group, string $user, ?string $byUser = null): ResponseInterface {
+		$credentials = $this->getAdminOrUserCredentials($byUser);
+		try {
+			$groupId = $this->featureContext->getAttributeOfCreatedGroup($group, "id");
+		} catch (Exception $e) {
+			$groupId = WebDavHelper::generateUUIDv4();
+		}
+		try {
+			$userId = $this->featureContext->getAttributeOfCreatedUser($user, "id");
+		} catch (Exception $e) {
+			$userId = WebDavHelper::generateUUIDv4();
+		}
```

Comment:
imho this logic is best suited for the step defn func. which supports non-exisent or tries to cases rather here. why doing such logic on this helper func?